Makefile

```diff
@@ -80,6 +82,13 @@ bin/manager: $(MANAGER_BIN_INPUTS)
 	go vet ./...
 	go build -o bin/manager main.go
 
+.PHONY: build-for-docker
+build-for-docker: bin/manager-linux-amd64 ## Build manager binary for docker image building.
+bin/manager-linux-amd64: $(MANAGER_BIN_INPUTS)
+	CGO_ENABLED=0 GOOS=linux GOARCH=amd64 \
```

Comment:
I'd like to have more confidence that cross-platform building locally, ostensibly on darwin, doesn't introduce differences that actually building in linux (debian in this case) env doesn't.  I don't see other CAPI providers doing this.  Do you have any research that supports the safety/reliability of this technique?

Reply:
I'm fine either way @wongni  if you do move back to the old way, we can just put this Dockerfile in hack and use it only for e2e.

Reply:
I didn't do any research about how reliable golang's cross-compilation is, but when I tested a CAPC image built in this way on my Macbook in a Linux VM it worked well.